src/ocdm/OpenCDMDecryptor.cpp

```diff
@@ -211,12 +212,14 @@ namespace CENCDecryptor {
                 opencdm_destruct_system(_system);
             }
 
-            Core::Singleton::Dispose();
+            if (_disposeInstance == true) {
```

Comment:
Do you know if there's a way to determine, if this app was launched by a `WPEProcess` here? The `WPEProcess` invokes the `Core::Singleton::Dispose()` when the process is torn down. Using this, we could contain this behavior to this function, instead of passing a flag through the interface

Reply:
I added the check based on proc entry status